### PR TITLE
fix: BUG FIXED 305

### LINKED ISSUE #305: BUGS: Solve the seeding Issue with Ingredients
Solve the seeding problem for 3000+ ingredinetns

### PR DESCRIPTION
**Pull request by MNSM92**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #305

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: apps/server/prisma/seed.ts [hunk 1]
```diff
@@ -1,47 +1,53 @@
...
     };
 
-    insertIngredients();
+    await insertIngredients(); // Await the insertion function
+
+    await prisma.$disconnect();
 }
 
-main().catch(e => {
+main().catch((e) => {
```
Comment (TanvirAlam) [inline apps/server/prisma/seed.ts]: We don't need this `process.exit(1);`?, they say it might create problem
please investigate, if this is needed

Comment (TanvirAlam) [reply apps/server/prisma/seed.ts]: we were cloing the conect when everything was done?
```
.finally(async () => {
    await prisma.$disconnect();
})
```

File: apps/server/prisma/seed.ts [hunk 2]
```diff
@@ -1,47 +1,53 @@
...
+            });
+            bar.tick();
+        } catch (error) {
+            console.error(`Error inserting ingredient: ${error.message}`);
+        }
     };
 
     const insertIngredients = async () => {
-        const promises = ingredients.map((ingredient: { name: any; imageName: any; }) => createIngredient(ingredient));
+        const promises = ingredients.map((ingredient :any) => createIngredient(ingredient));
```
Comment (TanvirAlam) [inline apps/server/prisma/seed.ts]: `any`? is bad

File: apps/server/prisma/seed.ts [hunk 3]
```diff
@@ -1,47 +1,53 @@
...
-            data: {
-                name: ingredient.name,
-                imageName: ingredient.imageName,
-            },
-        });
-        bar.tick();
+        total: totalIngredients,
+    });
+
+    const createIngredient = async (ingredient : any) => {
```
Comment (TanvirAlam) [inline apps/server/prisma/seed.ts]: `any` is bad

**Review by TanvirAlam — CHANGES_REQUESTED** [review]
LGTM, I asked few questions

File: apps/server/prisma/seed.ts (hunk 1 above)
Comment (MNSM92) [reply apps/server/prisma/seed.ts]: I actually didn't wanted to close the connection, to find the bug. But the bug is not actually inside the seed.ts but in .env

**Review by Rakibs511 — APPROVED** [review]

**Review by TanvirAlam — APPROVED** [review]
LGTM

**Pull request merged**